### PR TITLE
Add support for LiteralFloat type

### PR DESCRIPTION
**Pull request by asudarsa**
In https://github.com/KhronosGroup/SPIRV-Headers/pull/348
We are trying to add a new literal type called LiteralFloat.
This PR aims to add support in SPIR-V Tools to support this new type.

Thanks

### CODE REVIEW

**Comment by s-perron** [comment]
This looks fine to me, but we cannot merge until the spirv-headers PR is merged, and we roll the dependencies.

**Review by dneto0 — APPROVED** [review]
Looks good to me.

The SPV_OPERAND_TYPE_TYPED_LITERAL_NUMBER does not fit here because that one requires type information, determined from immediately surrounding context.
..TYPED_LITERAL_NUMBER is used for OpConstant, OpSpecContant, and OpSwitch. The first two are used for float literals too, but have the float type right beside them.

Since this doesn't have an immediate context, a special case for float is ok.

Future readers should keep that in mind.

**Pull request merged**
